### PR TITLE
Improve list user permissions func on perms store

### PR DESCRIPTION
**Pull request by thenamankumar**
In response to feedback collected from PR: https://github.com/sourcegraph/sourcegraph/pull/48283

## Test plan

updated unit tests. and manually tests graphql api

### CODE REVIEW

File: internal/database/repos.go [hunk 1]
```diff
@@ -474,107 +474,13 @@ var repoColumns = []string{
 	"(SELECT json_object_agg(key, value) FROM repo_kvps WHERE repo_kvps.repo_id = repo.id)",
 }
 
-type RepoRowScanner = func(rows *sql.Rows, repo *types.Repo, metadata *json.RawMessage, blocked *dbutil.NullJSONRawMessage, kvps *RepoKVPs, sources *dbutil.NullJSONRawMessage) error
+func scanRepo(logger log.Logger, rows *sql.Rows, r *types.Repo) (err error) {
```
Comment (thenamankumar) [inline internal/database/repos.go:477]: Changes to this file in the previous PR are completely reverted here as they are no longer required.

File: enterprise/internal/database/perms_store.go [hunk 2]
```diff
@@ -2109,32 +2108,12 @@ func (s *permsStore) ListUserPermissions(ctx context.Context, userID int32, args
...
-				blocked,
-				kvps,
-				sources,
-				&dbutil.NullTime{Time: &updatedAt},
-				&reason,
-			)
-		})
-
-		if err := scanRepo(s.logger, rows, &repo); err != nil {
+		if err := rows.Scan(
```
Comment (sashaostrikov) [inline enterprise/internal/database/perms_store.go:2111]: nice, looks so much cleaner now! 🎉

File: enterprise/internal/database/perms_store_test.go [hunk 3]
```diff
@@ -3427,170 +3427,175 @@ func setupTestPerms(t *testing.T, db database.DB, clock func() time.Time) *perms
...
 
-func TestPermsStore_ListUserPermissions(t *testing.T) {
-	logger := logtest.Scoped(t)
-	testDb := dbtest.NewDB(logger, t)
-	db := database.NewDB(logger, testDb)
-	s := perms(logger, db, clock)
-	ctx := context.Background()
-	t.Cleanup(func() {
-		cleanupPermsTables(t, s)
+func testPermsStore_ListUserPermissions(db database.DB, logger log.Logger) func(*testing.T) {
```
Comment (sashaostrikov) [inline enterprise/internal/database/perms_store_test.go:3430]: you can safely remove logger from parameter list and use `logtest.Scoped(t)` below

File: enterprise/internal/database/perms_store_test.go [hunk 4]
```diff
@@ -3427,170 +3427,175 @@ func setupTestPerms(t *testing.T, db database.DB, clock func() time.Time) *perms
...
+	return func(t *testing.T) {
+		s := perms(logger, db, clock)
```
Comment (sashaostrikov) [inline enterprise/internal/database/perms_store_test.go:3432]:
```suggestion
		s := perms(logtest.Scoped(t), db, clock)
```

File: enterprise/internal/database/perms_store_test.go [hunk 5]
```diff
@@ -3427,170 +3427,175 @@ func setupTestPerms(t *testing.T, db database.DB, clock func() time.Time) *perms
...
-				{
-					RepoId: 5,
-					Reason: UserRepoPermissionReasonUnrestricted,
+		tests := []listUserPermissionsTest{
+			{
+				Name:   "TestNonSiteAdminUser",
+				UserID: 555,
+				WantResults: []*listUserPermissionsResult{
+					{
+						// private repo but have access via user_permissions
```
Comment (sashaostrikov) [inline enterprise/internal/database/perms_store_test.go:3502]: like this comments 👍🏻

File: enterprise/internal/database/perms_store_test.go [hunk 6]
```diff
@@ -3427,170 +3427,175 @@ func setupTestPerms(t *testing.T, db database.DB, clock func() time.Time) *perms
...
+				if err != nil {
+					t.Fatal(err)
+				}
 
-		for index, result := range results {
-			if diff := cmp.Diff(test.WantResults[index], &listUserPermissionsResult{RepoId: int32(result.Repo.ID), Reason: result.Reason}); diff != "" {
-				t.Fatalf("Test %d. Results %d mismatch (-want +got):\n%s", tIndex, index, diff)
-			}
+				if len(test.WantResults) != len(results) {
+					t.Fatalf("Test %d. Results mismatch. Want: %d Got: %d", tIndex, len(test.WantResults), len(results))
```
Comment (sashaostrikov) [inline enterprise/internal/database/perms_store_test.go:3584]: nit: we can skip test numbers now, since we have test names 😎

**Review by sashaostrikov — APPROVED** [review]
LGTM! Left a couple of suggestions in tests

File: enterprise/internal/database/perms_store_test.go (hunk 6 above)
Comment (thenamankumar) [reply enterprise/internal/database/perms_store_test.go:3584]: removed

File: enterprise/internal/database/perms_store_test.go (hunk 4 above)
Comment (thenamankumar) [reply enterprise/internal/database/perms_store_test.go:3432]: resolved

File: enterprise/internal/database/perms_store_test.go (hunk 3 above)
Comment (thenamankumar) [reply enterprise/internal/database/perms_store_test.go]: done

**Pull request merged**